text/0000-generic-dependent-consts.md

```diff
@@ -0,0 +1,291 @@
+- Feature Name: generic_dependent_consts
+- Start Date: 2015-03-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Allow consts declared in functions to depend on function type parameters. Since
+this raises some issues in match and type checks, use an conservative approach
+that allows only minimal use of such consts when a constant expression is
+required.
+
+# Motivation
+
+Consider the following line of code in a non-generic context:
+
+```rust
+let a: [u8; T::N + T::N] = [0u8; 2*T::N];
```

Comment:
I don't quite understand. If I write something like
```
trait HasN {
    const N: usize;
}
fn f<T: HasN>() {
    let a = [0u8; 2 * T::N];
    println!("{}", a[0]);
}
```
shouldn't the type `[u8; 2 * T::N]` be "evaluated" during instantiation of `f` leading to an error? And there's no runtime here, because `f` can't even be instantiated with `T::N > usize::MAX / 2`.